### PR TITLE
Issue #1988 alternative

### LINKED ISSUE #1988: Date range on Step 2 not working
## Steps to reproduce
1. Select AusCPR and go to Step 2
2. Enter the date range 2010/01/11 in both the "from" and "to" fields

## What happens 
The map is empty

## What I expect to happen
Data from the entire day (of the 11th) displayed on the map - not just data at 00-UTC

### PR DESCRIPTION
**Pull request by dnahodil**
This is another way to fix #1988

### CODE REVIEW

**Comment by dnahodil** [comment]
@pmbohm this is the sort of thing I was thinking in my comment (https://github.com/aodn/aodn-portal/pull/2004#issuecomment-145403435).

File: web-app/js/portal/filter/ui/DateFilterPanel.js [hunk 1]
```diff
@@ -101,7 +101,14 @@ Portal.filter.ui.DateFilterPanel = Ext.extend(Portal.filter.ui.BaseFilterPanel,
 
         this.filter.setValue({
             fromDate: this.fromDate.getValue(),
-            toDate: this.toDate.getValue()
+            toDate: this._endOfDay(this.toDate.getValue())
         });
+    },
+
+    _endOfDay: function(date) {
```
Comment (jkburges) [inline web-app/js/portal/filter/ui/DateFilterPanel.js]: Can use http://momentjs.com/docs/#/manipulating/end-of/ ?

Except this is a javascript date, not a moment. But maybe we can still use it?

Comment (pmbohm) [inline web-app/js/portal/filter/ui/DateFilterPanel.js]: I tried that function. The results were not really what is wanted as i dont think it handles the fact that we want the last moment of a UTC day not the last second of the date in client timezone expressed as UTC

Comment (pmbohm) [inline web-app/js/portal/filter/ui/DateFilterPanel.js]: I am a fan of using moment though
```
        var momentDate = moment.utc(date);
        if (date && momentDate.isValid()) {
            return momentDate.add(1, 'd').toDate();
        }
```

File: web-app/js/portal/filter/ui/DateFilterPanel.js [hunk 2]
```diff
@@ -101,7 +101,13 @@ Portal.filter.ui.DateFilterPanel = Ext.extend(Portal.filter.ui.BaseFilterPanel,
...
+        if (date) {
+            return moment(date).endOf('day').toDate();
```
Comment (dnahodil) [inline web-app/js/portal/filter/ui/DateFilterPanel.js]: How's that? or is it too much for one line?

**Comment by dnahodil** [comment]
@jkburges changes made for suggestion

File: web-app/js/portal/filter/ui/DateFilterPanel.js (hunk 2 above)
Comment (jkburges) [inline web-app/js/portal/filter/ui/DateFilterPanel.js]: Is this going to work w.r.t. https://github.com/aodn/aodn-portal/pull/2016/files#r41110785 ?

Comment (dnahodil) [inline web-app/js/portal/filter/ui/DateFilterPanel.js]: It seemed to work as I expected it to. I did a test download on the Argo collection and the downloaded data was all within the correct day.

**Pull request merged**